<!--

  DO NOT EDIT.

  This markdown file was autogenerated using a mixture of the following files as the source of truth for its data:
  - ../../src/rules/sort-imports-metadata.ts
  - ../../tests/rules/sort-imports-metadata/cases.ts

  In order to update this file, it is therefore those files which need to be updated, as well as potentially the generator script:
  - ../../../../tools/scripts/generate-rule-docs.ts

-->

<br>

# `@angular-eslint/sort-imports-metadata`

Ensures ASC alphabetical order for import arrays for easy visual scanning

- Type: suggestion
- üîß Supports autofix (`--fix`)

<br>

## Rule Options

The rule accepts an options object with the following properties:

```ts
interface Options {
  /**
   * A string with a BCP 47 language tag.
   *
   * Default: `"en-US"`
   */
  locale?: string;
}

```

<br>

## Usage Examples

> The following examples are generated automatically from the actual unit tests within the plugin, so you can be assured that their behavior is accurate based on the current commit.

<br>

<details>
<summary>‚ùå - Toggle examples of <strong>incorrect</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-imports-metadata": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Component({
  standalone: true,
  selector: 'test-component',
  imports: [aModule, bModule, dModule, cModule],
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
export class TestComponent { }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-imports-metadata": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Directive({
  standalone: true,
  selector: 'test-directive',
  imports: [aModule, bModule, dModule, cModule],
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
export class TestDirective { }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-imports-metadata": [
      "error"
    ]
  }
}
```

<br>

#### ‚ùå Invalid Code

```ts
@Pipe({
  standalone: true,
  selector: 'test-pipe',
  imports: [aModule, bModule, dModule, cModule],
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
})
export class TestPipe { }
```

</details>

<br>

---

<br>

<details>
<summary>‚úÖ - Toggle examples of <strong>correct</strong> code for this rule</summary>

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-imports-metadata": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Component({
  standalone: true,
  selector: 'test-component',
  imports: [aModule, bModule, cModule, dModule],
})
export class TestComponent { }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-imports-metadata": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Directive({
  standalone: true,
  selector: 'test-directive',
  imports: [aModule, bModule, cModule, dModule],
})
export class TestDirective { }
```

<br>

---

<br>

#### Default Config

```json
{
  "rules": {
    "@angular-eslint/sort-imports-metadata": [
      "error"
    ]
  }
}
```

<br>

#### ‚úÖ Valid Code

```ts
@Pipe({
  standalone: true,
  selector: 'test-pipe',
  imports: [aModule, bModule, cModule, dModule],
})
export class TestPipe { }
```

</details>

<br>
